Nos/Views/Modifiers/ReportMenuModifier.swift

```diff
@@ -182,7 +202,34 @@ struct ReportMenuModifier: ViewModifier {
             }
         }
     }
-    
+
+    func unmute(author: Author) {
+        Task {
+            do {
+                try await author.unmute(viewContext: viewContext)
+            } catch {
+                Log.error(error.localizedDescription)
+            }
+        }
+    }
+
+    /// Determines the visibility status for a flagged account and applies the appropriate action.
+    /// - Parameter completion: A closure that is executed once the necessary actions are complete.
+    private func determineFlaggedAccoutVisibility(completion: () -> Void) {
+        if let author = reportedObject.author {
+            guard case .visibility(let visibilityCategory) = selectedVisibilityOption?.category else { return }
+
+            if visibilityCategory == .mute {
+                guard !author.muted else { return }
+                mute(author: author)
+            } else {
+                guard author.muted else { return }
+                unmute(author: author)
+            }
+            completion()
```

Comment:
Ah, I just noticed that `completion()` may be called before the `mute` or `unmute` is finished. If we want to make sure those have finished, we'll probably want them to be async instead of having a `Task` inside them. That might affect other things; I'd be happy to look and talk through it if you'd like.